crates/ibc/src/test_utils.rs

```diff
@@ -73,113 +73,24 @@ pub fn get_dummy_transfer_module() -> DummyTransferModule {
     let ibc_store = Arc::new(Mutex::new(MockIbcStore::default()));
     DummyTransferModule { ibc_store }
 }
+
+pub fn get_dummy_transfer_context() -> DummyTransferContext {
+    let ibc_store = Arc::new(Mutex::new(MockIbcStore::default()));
+    DummyTransferContext { ibc_store }
+}
+
 #[derive(Debug)]
-pub struct DummyTransferModule {
+pub struct DummyTransferContext {
     ibc_store: Arc<Mutex<MockIbcStore>>,
 }
```

Comment:
Yes this is the messiest part of the solution. The core problem is that the previous way it was implemented [downcasted](https://github.com/cosmos/ibc-rs/blob/127f12936b678bfe979f8ad12e4f170c89b8ff40/crates/ibc/src/core/handler.rs#L475-L477) the `Module` to a `DummyTransferModule`, and we can on longer do that because downcasting requires `core::any::Any`, which requires `'static`. See #491 for how I think this could be improved.

I ended up going for this because it worked. I think there was an issue when implementing on the `MockContext`, because we still need the traits for the callback handlers... But then again there might be a way to make it work.

Note that I find that the shared store architecture is confusing because it couples our types in confusing ways; the whole thing is in need of a refactor IMO.